shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-oracle/src/main/antlr4/imports/oracle/DDLStatement.g4

```diff
@@ -941,9 +945,51 @@ addListPartitionClause
     ;
 
 dropTablePartition
-    : DROP partitionExtendedNames
+    : DROP partitionExtendedNames (updateIndexClauses parallelClause?)?
     ;
 
 partitionExtendedNames
-    : (PARTITION | PARTITIONS) partition
+    : (PARTITION | PARTITIONS) (partitionName | partitionForClauses) (COMMA_ (partitionName | partitionForClauses))*
+    ;
+
+partitionForClauses
+    : FOR LP_ partitionKeyValue (COMMA_ partitionKeyValue)* RP_
+    ;
+
+updateIndexClauses
+    : updateGlobalIndexClause | updateAllIndexesClause
+    ;
+
+updateGlobalIndexClause
+    : (UPDATE | INVALIDATE) GLOBAL INDEXES
+    ;
+
+updateAllIndexesClause
+    : UPDATE INDEXES
+    (LP_ indexName LP_ (updateIndexPartition | updateIndexSubpartition)
+    (COMMA_ indexName LP_ (updateIndexPartition | updateIndexSubpartition) RP_)* RP_)?
+    ;
+
+updateIndexPartition
+    : indexPartitionDesc indexSubpartitionClause?
+    (COMMA_ indexPartitionDesc indexSubpartitionClause?)*
+    ;
+
+indexPartitionDesc
+    : PARTITION
+    (partitionName
+    ((segmentAttributesClause | indexCompression) | PARAMETERS LP_ SQ_ odciParameters SQ_ RP_ )?
```

Comment:
The rule is not correct.
![image](https://user-images.githubusercontent.com/22066046/114637233-6a5aec00-9cfb-11eb-8438-22e27853f1bc.png)

Reply:
Yes! It should be `(segmentAttributesClause | indexCompression)+`. 

Actually this relates to my question in #10016. Before when I add this `+`, it give me the following error
`rule XXX contains an optional block with at least one alternative that can match an empty string`

I notice that you changed the rule `physicalAttributesClause` from
`(PCTFREE NUMBER_ | PCTUSED NUMBER_ | INITRANS NUMBER_ | storageClause)*` to 
`(PCTFREE NUMBER_ | PCTUSED NUMBER_ | INITRANS NUMBER_ | storageClause)+` to avoid this error. 

However, I think the former rule is more accurate as it could also be empty. Should we then use `physicalAttributesClause?` in rules which contains `physicalAttributesClause` to allow empty input as well?
![image](https://user-images.githubusercontent.com/66914151/114790430-07ae3280-9d85-11eb-9db7-55501a535ce8.png)

Reply:
“(PCTFREE NUMBER_ | PCTUSED NUMBER_ | INITRANS NUMBER_ | storageClause)* ” is more suitable.

Reply:
“(PCTFREE NUMBER_ | PCTUSED NUMBER_ | INITRANS NUMBER_ | storageClause)* ” is more suitable. @Liangda-w